homeassistant/components/kef/media_player.py

```diff
@@ -304,3 +391,69 @@ def icon(self):
     async def async_media_next_track(self):
         """Send next track command."""
         await self._speaker.next_track()
+
+    async def update_dsp(self):
+        """Update the DSP settings."""
+        mode = await self._speaker.get_mode()
+        self._dsp = dict(
+            **mode._asdict(),
+            desk_db=await self._speaker.get_desk_db(),
+            wall_db=await self._speaker.get_wall_db(),
+            treble_db=await self._speaker.get_treble_db(),
+            high_hz=await self._speaker.get_high_hz(),
+            low_hz=await self._speaker.get_low_hz(),
+            sub_db=await self._speaker.get_sub_db(),
```

Comment:
This will run it all in sequence. If you pass it to `asyncio.gather` it will run in parallel.

Reply:
The speaker doesn't handle parallel requests.

I implemented it via locks, so using gather would still work, however, there's no benefit of doing it then I think.